src/main/java/frc/robot/subsystems/Revolver.java

```diff
@@ -56,16 +41,27 @@ public void setAll(double drumSpeed, double feedWheelSpeed) {
     setFeedWheel(feedWheelSpeed);
   }
 
+  public void resetReps(){
+    m_drumReps = 0;
+  }
+
   /**
    * Sets the speed of the drum
    * @param speed the speed to run the drum at as a percentage from -1 to 1
    */
   public void setDrum(double speed) {
     m_drumMotor.set(speed);
-    if (m_pdp.getCurrent(RobotMap.PDP.DRUM_PORT) > Constants.REVOLVER_DRUM_CURRENT_THRESHOLD) {
-      m_repsAboveDrum++;
-    } else {
-      resetDrumReps();
+
+    if((speed==0) || (speed!=0 && m_drumMotor.getEncoder().getVelocity()!=0)){
+      resetReps();
+    }
+
+    if (speed > 0 && m_drumMotor.getEncoder().getVelocity()==0.0){
```

Comment:
Velocity threshold to increment reps should be very low, but not zero, and should be moved to constants